src/main/java/org/opensearch/neuralsearch/processor/AgenticQueryTranslatorProcessor.java

```diff
@@ -124,7 +129,29 @@ private void executeAgentAsync(
             try {
                 log.info("Generated Query: [{}]", agentResponse);
 
-                // TODO: Add query validations
+                // Validate response size to prevent memory exhaustion
+                if (agentResponse == null) {
+                    String errorMessage = String.format(
+                        Locale.ROOT,
+                        "Agentic search failed - Null response from agent - Agent ID: [%s], Query: [%s]",
+                        agentId,
+                        agenticQuery.getQueryText()
+                    );
+                    throw new IllegalArgumentException(errorMessage);
+                }
+
+                if (agentResponse.length() > MAX_AGENT_RESPONSE_SIZE) {
+                    String errorMessage = String.format(
+                        Locale.ROOT,
+                        "Agentic search blocked - Response size exceeded limit - Agent ID: [%s], Size: [%d], Query: [%s]. Maximum allowed size is %d characters.",
+                        agentId,
+                        agentResponse.length(),
+                        agenticQuery.getQueryText(),
+                        MAX_AGENT_RESPONSE_SIZE
+                    );
+                    throw new IllegalArgumentException(errorMessage);
+                }
```

Comment:
what is current behavior in case user input isn't a valid json?

Reply:
this is the response from execute agent and not user input, it will return a json with the default prompt